### PR TITLE
Animate progress bar changes in data collection flow

### LINKED ISSUE #1794: Animate progress bar to data collection flow

### PR DESCRIPTION
**Pull request by JSunde**
Fixes #1794

https://github.com/google/ground-android/assets/9056308/76454515-d849-49bc-a70d-ef17c391359f

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt [hunk 1]
```diff
@@ -99,12 +101,24 @@ class DataCollectionFragment : Hilt_DataCollectionFragment(), BackPressListener
...
-    progressBar.progress = index
+
+    progressBar.clearAnimation()
+
+    val progressAnimator = ValueAnimator.ofInt(progressBar.progress, index * PROGRESS_SCALE)
+    progressAnimator.duration = 400L
+    progressAnimator.interpolator = FastOutSlowInInterpolator()
+
+    progressAnimator.addUpdateListener {
+      val updatedProgress = it.animatedValue as Int
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt:117]: Can this variable be inlined?

**Review by gino-m — CHANGES_REQUESTED** [review]
One nit, otherwise LG

File: ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt]: Done!

**Review by gino-m — APPROVED** [review]

**Pull request merged**